### ISSUE TITLE
Need to polish null char filled strings.

### ISSUE THREAD

**Issue by donggas90** [sharpdx/SharpDX#806]
DXGI Adapter and DXGI Ouput has its Description for each and the desc has its HW name.
That field size is fixed in native world. Therefore, after valid chars, fill with null chars.
But SharpDX do not handle these null chars currently. Then, the name string looks like this "name\0\0\0\0".
I think, just mashalling the fixed size of memory. Might be, it's not expected result in SharpDX users side such as me.
However, if you want preserve native clarity, please add some comments to them.
I was so confused by a bug that because of the null chars.

**Comment by xoofx** [comment]
Indeed, this is a bug. String marshaling had to be refactored a long time ago (when having to provide SharpDX for Metro/Windows 8) and maybe got break at this time.

**Comment by xoofx** [comment]
Pull request are welcome, as I don't have very limited spare time dedicated to SharpDX.

**Comment by donggas90** [comment]
@xoofx I'm sorry to hear that, but don't know, how the SharpDX converting C++ code to C# in deeply. Might it works by SharpGen? Known about mapping XMLs are exist but don't know, how to use them such like syntax. Therefore, I didn't change the source code at all. Instead, just added a function to handle it. 
```
public static string RefineNullFilledString(string rawString)
{
    System.Text.StringBuilder builder = new System.Text.StringBuilder(rawString.Length);
    foreach (char c in rawString)
    {
        if (c != '\0')
        {
            builder.Append(c);
        }
        else
        {
            break;
        }
    }
    return builder.ToString();
}
```
If you provide some references about your converting system, I'll try fixing this to appreciate your efforts.
